### PR TITLE
minipro: 0.5 -> 0.6

### PR DESCRIPTION
**Pull request by hzeller**
###### Description of changes

Update version tagged upstream.
https://gitlab.com/DavidGriffith/minipro/-/tags/0.6

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by hzeller** [comment]
The failure in the CI looks unrelated: some full disk on the CI machine.

File: pkgs/tools/misc/minipro/default.nix [hunk 1]
```diff
@@ -7,23 +7,25 @@
...
   nativeBuildInputs = [ pkg-config ];
   buildInputs = [ libusb1 ];
   makeFlags = [
+    "VERSION=${version}"
     "PREFIX=$(out)"
     "UDEV_DIR=$(out)/lib/udev"
     "COMPLETIONS_DIR=$(out)/share/bash-completion/completions"
     "PKG_CONFIG=${pkg-config}/bin/${pkg-config.targetPrefix}pkg-config"
     "CC=${stdenv.cc.targetPrefix}cc"
+    "CFLAGS=-O2"
```
Comment (bmwalters) [inline pkgs/tools/misc/minipro/default.nix:28]: why override cflags?

**Review by bmwalters — APPROVED** [review]

File: pkgs/tools/misc/minipro/default.nix (hunk 1 above)
Comment (hzeller) [reply pkgs/tools/misc/minipro/default.nix:28]: The default `CFLAGS` has no optimization  `-O0`. It looks like it is for convenience of development running a debugger and such, but not really something one would do for a production binary.

Comment (bmwalters) [reply pkgs/tools/misc/minipro/default.nix:28]: makes sense; my only concern is that upstream enables `-Wall -Wextra -W -Wno-sign-compare -Wno-unused-parameter` so by overriding cflags we're building with a different set of error flags

Comment (hzeller) [reply pkgs/tools/misc/minipro/default.nix:28]: These warnings are more important for the developers (in fact, I sent some of these in a pull request upstream to minipro) to make sure that the code is clean. But for building the release binary they are not providing any benefit.

**Comment by hzeller** [comment]
I guess we need to find more approvers for it to get merged, or how does this process work ?

**Comment by ck3d** [comment]
Result of `nixpkgs-review pr 191741` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>minipro</li>
  </ul>
</details>

**Pull request merged**